src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -0,0 +1,99 @@
+package wasm
+
+import (
+	"encoding/binary"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"vectorized/pkg/kafka"
+	"vectorized/pkg/utils"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewDeployCommand(
+	fs afero.Fs, createProducer func(bool, int32) (sarama.SyncProducer, error),
+) *cobra.Command {
+	var (
+		description string
+	)
+
+	command := &cobra.Command{
+		Use:	"deploy <path>",
+		Short:	"deploy inline WASM function",
+		Args:	cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			path := args[0]
+			fullFileName := filepath.Base(path)
+			fileExt := filepath.Ext(fullFileName)
+			// validate file extension, just allow js extension
+			if fileExt != ".js" {
+				return fmt.Errorf("can't deploy '%s': only .js files are supported.", path)
+			}
+			fileName := strings.Replace(fullFileName, fileExt, "", 1)
+			fileContent, err := utils.ReadFileLines(fs, path)
+			if err != nil {
+				return err
+			}
+			var bytes []byte
+			for _, v := range fileContent {
+				bytes = append(bytes, []byte(v)...)
+			}
+			// create producer
+			producer, err := createProducer(false, -1)
+
+			if err != nil {
+				return err
+			}
+			return deploy(
+				fileName,
+				bytes,
+				description,
+				producer,
+			)
+		},
+	}
+
+	command.Flags().StringVar(
+		&description,
+		"description",
+		"",
+		"description about what or how wasm function do, in order to "+
+			"give some information to another developers",
+	)
+
+	return command
+}
+
+func deploy(
+	fileName string,
+	fileContent []byte,
+	description string,
+	producer sarama.SyncProducer,
+) error {
+	// createMessage
+	messageContent := createCoprocessorDeployMessage(fileContent, description)
+	// publish message
+	error := kafka.PublishMessage(messageContent, "deploy/"+fileName, producer, kafka.CoprocessorTopic)
+	if error != nil {
+		return fmt.Errorf("Error on deploy %s.js, message error: %s, please check "+
+			"your redpanda instance ", fileName, error)
+	}
+	return nil
+}
+
+func createCoprocessorDeployMessage(content []byte, description string) []byte {
+	// create buffer for filename and content size bytes
+	sizeFileContent := make([]byte, 64)
+	sizeDescription := make([]byte, 64)
+	binary.LittleEndian.PutUint64(sizeFileContent, uint64(len(content)))
```

Comment:
hmm... this is a public interface. We should have a chat about this. 

We need to think on the format. my hunch is a `json key` and the bytes as value. kafka is a key=value pair tuples. 
the batch format also has header information we can use.

Reply:
@senior7515 could you give more details? we need to put in that message the coprocessor file content.

Reply:
new format: 

```
{
	Key: wasm_name
	Header: {
		Action: "deploy"
		Description: wasm_descripton
                checksum: sha256(contentFile) 
        }
        Message: wasm_content_file
}
```